include/adc.h

```diff
@@ -26,121 +26,303 @@ extern "C" {
  * @{
  */
 
+
+/** @brief ADC channel gain factors. */
+enum adc_gain {
+	ADC_GAIN_1_6, /**< x 1/6. */
+	ADC_GAIN_1_5, /**< x 1/5. */
+	ADC_GAIN_1_4, /**< x 1/4. */
+	ADC_GAIN_1_3, /**< x 1/3. */
+	ADC_GAIN_1_2, /**< x 1/2. */
+	ADC_GAIN_1,   /**< x 1. */
+	ADC_GAIN_2,   /**< x 2. */
+	ADC_GAIN_3,   /**< x 3. */
+	ADC_GAIN_4,   /**< x 4. */
+	ADC_GAIN_8,   /**< x 8. */
+	ADC_GAIN_16,  /**< x 16. */
+	ADC_GAIN_32,  /**< x 32. */
+	ADC_GAIN_64,  /**< x 64. */
+};
+
+/** @brief ADC references. */
+enum adc_reference {
+	ADC_REF_VDD_1,     /**< VDD. */
+	ADC_REF_VDD_1_2,   /**< VDD/2. */
+	ADC_REF_VDD_1_3,   /**< VDD/3. */
+	ADC_REF_VDD_1_4,   /**< VDD/4. */
+	ADC_REF_INTERNAL,  /**< Internal. */
+	ADC_REF_EXTERNAL0, /**< External, input 0. */
+	ADC_REF_EXTERNAL1, /**< External, input 1. */
+};
+
+#define ADC_ACQ_TIME_DEFAULT       0
+#define ADC_ACQ_TIME_TICKS         (1u)
+#define ADC_ACQ_TIME_MICROSECONDS  (2u)
+#define ADC_ACQ_TIME_NANOSECONDS   (3u)
+#define ADC_ACQ_TIME(unit, value)  (((unit) << 14) | \
+				    ((value) & ((1u << 14)-1)))
+#define ADC_ACQ_TIME_UNIT(time)    (((time) >> 14) & 3u)
+#define ADC_ACQ_TIME_VALUE(time)   ((time) & ((1u << 14)-1))
+
 /**
- * @brief ADC driver Sequence entry
- *
- * This structure defines a sequence entry used
- * to define a sample from a specific channel.
+ * @brief Structure for specifying the configuration of an ADC channel.
  */
-struct adc_seq_entry {
-	/** Clock ticks delay before sampling the ADC. */
-	s32_t sampling_delay;
+struct adc_channel_cfg {
+	/** Gain selection. */
+	enum adc_gain gain;
+
+	/** Reference selection. */
+	enum adc_reference reference;
+
+	/**
+	 * Acquisition time.
+	 * Use the ADC_ACQ_TIME macro to compose the value for this field or
+	 * pass ADC_ACQ_TIME_DEFAULT to use the default setting for a given
+	 * hardware (e.g. when the hardware does not allow to configure the
+	 * acquisition time).
+	 * Particular drivers do not necessarily support all the possible units.
+	 * Value range is 0-16383 for a given unit.
+	 */
+	u16_t acquisition_time;
+
+	/** Channel identifier (0-31). */
+	u8_t channel_id   : 5;
+
+	/** Channel type: single-ended or differential. */
+	bool differential : 1;
```

Comment:
Keep u8_t here, I am not entirely sure compiler will properly pack it with previous byte.